keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -1189,9 +1217,76 @@ based on storage capacity:
   to available storage and thus could run on a new node, the
   simulation may decide otherwise.
 
-It may be possible to solve this by pre-configuring some information
-(local storage capacity of future nodes and their CSI topology). This
-needs to be explored further.
+This gets further complicated by the independent development of CSI drivers,
+autoscaler, and cloud provider: autoscaler and cloud provider don't know which
+kinds of volumes a CSI driver will be able to make available on nodes because
+that logic is implemented inside the CSI driver. The CSI driver doesn't know
+about hardware that hasn't been provisioned yet and doesn't know about
+autoscaling.
+
```

Comment:
I think the the concern here is signing off on this as an agreed upon solution. Instead let's word this to indicate that further design and investigation is still needed to solve these problems:
- How a CSI driver can report capacity on a pristine, simulated node
- How the autoscaler can provide a CSI driver with the storage information it needs to be able to model capacity based on the environment it's modeling.
- How to improve the batch scheduling use case

And then park the existing discussion under "Alternatives" (don't want to lose the great discussion that's happening here).

Reply:
> How the autoscaler can provide a CSI driver with the storage information it needs to be able to model capacity based on the environment it's modeling.

Isn't it meant to be the other way around: the CSI driver provides information to the autoscaler and the autoscaler does the modeling?

> And then park the existing discussion under "Alternatives" (don't want to lose the great discussion that's happening here).

I've pushed an update that under "drawbacks" describes the problem and that further work is needed, then under "alternatives" the description of the current prototype.

Reply:
> Isn't it meant to be the other way around: the CSI driver provides information to the autoscaler and the autoscaler does the modeling?

I think there may need to be information passed both ways. In order for a CSI driver to estimate a capacity without being able to actually inspect a node, it's going to need information about the future node, such as how many disks are going to be there and their capacities.